starknet_py/contract.py

```diff
@@ -382,26 +452,75 @@ async def estimate_fee(
         *,
         nonce: Optional[int] = None,
     ) -> EstimatedFee:
+        tx = await self.get_account.sign_invoke_v1_transaction(
+            calls=self, nonce=nonce, max_fee=0
+        )
+        estimate_tx = await self.get_account.sign_for_fee_estimate(transaction=tx)
+
+        estimated_fee = await self._client.estimate_fee(
+            tx=estimate_tx,
+            block_hash=block_hash,
+            block_number=block_number,
+        )
+
+        assert isinstance(estimated_fee, EstimatedFee)
+        return estimated_fee
+
+
+@add_sync_methods
+@dataclass
+class PreparedFunctionInvokeV3(PreparedFunctionInvoke):
+    """
+    Prepared date to send an InvokeV3 transaction.
+    """
+
+    l1_resource_bounds: Optional[ResourceBounds]
+
+    async def invoke(
+        self,
+        l1_resource_bounds: Optional[ResourceBounds] = None,
+        auto_estimate: bool = False,
+        *,
+        nonce: Optional[int] = None,
+    ) -> InvokeResult:
         """
-        Estimate fee for prepared function call.
+        Send an Invoke transaction version 3 for a prepared data.
 
-        :param block_hash: Estimate fee at specific block hash.
-        :param block_number: Estimate fee at given block number
-            (or "latest" / "pending" for the latest / pending block), default is "pending".
+        :param l1_resource_bounds: Max amount and max price per unit of L1 gas (in Fri) used when executing
+            this transaction.
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs.
```

Comment:
```suggestion
        :param auto_estimate: Use automatic fee estimation; not recommended as it may lead to high costs.
```
or 
```suggestion
        :param auto_estimate: Use automatic fee estimation (not recommended, as it may lead to high costs).
```
(both can be found in the codebase)